src/udev/udev-event.c

```diff
@@ -549,9 +549,11 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
 
         l = read(fd, p, size - 1);
         if (l < 0) {
-                if (errno != EAGAIN)
+                if (errno != EAGAIN) {
+                        sd_event_source_set_enabled(s, SD_EVENT_ONESHOT);
```

Comment:
on EAGAIN the event source should be reenabled, but on anything else it should not be

Reply:
Yes indeed, I got mixed up with the sense of this check, will fix